server/Arcadia.Assistant.Web/Download/DownloadActor.cs

```diff
@@ -0,0 +1,186 @@
+﻿namespace Arcadia.Assistant.Web.Download
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.IO.Compression;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Threading.Tasks;
+
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Serialization;
+
+    using Akka.Actor;
+    using Akka.Event;
+
+    using Arcadia.Assistant.Web.Configuration;
+
+    public class DownloadActor : UntypedActor, ILogReceive
+    {
+        private readonly IDownloadApplicationSettings downloadApplicationSettings;
+        private readonly IHttpClientFactory httpClientFactory;
+
+        private readonly ILoggingAdapter logger = Context.GetLogger();
+
+        public DownloadActor(IDownloadApplicationSettings downloadApplicationSettings, IHttpClientFactory httpClientFactory)
+        {
+            this.downloadApplicationSettings = downloadApplicationSettings;
+            this.httpClientFactory = httpClientFactory;
+
+            Context.System.Scheduler.ScheduleTellRepeatedly(
+                TimeSpan.Zero,
+                TimeSpan.FromMinutes(downloadApplicationSettings.DownloadBuildIntervalMinutes),
+                this.Self,
+                RefreshApplicationBuilds.Instance,
+                this.Self);
+        }
+
+        protected override void OnReceive(object message)
+        {
+            switch (message)
+            {
+                case RefreshApplicationBuilds _:
+                    this.DownloadApplicationBuilds()
```

Comment:
I don't like that you might start another download while first is in progress. You probably need to have Become/Unbecome logic. On other it's highly unlikely :)

Reply:
Probably, it won't, because refresh period will be measured in tens minutes or even hours. But as you wish.

Reply:
take a look at the result, if it's ugly - don't commit it :)

Reply:
Done.